src/userflow.ts

```diff
@@ -206,6 +207,30 @@ interface Deferred {
 // empty object instead of `window`.
 var w: WindowWithUserflow = typeof window === 'undefined' ? ({} as any) : window
 var userflow = w.userflow
+
+// indicates if the history API's pushState and replaceState are patched with custom event emitters
+w.__userflowStatePatched = false
+
+// patch the history API's
+// pushState method to emit userflow:pushstate and
+// replaceState method to emit userflow:replacestate
+const originalPushState = window.history.pushState
+const originalReplaceState = window.history.replaceState
+
+w.history.pushState = (...args) => {
+  const userflowPushStateEvent = new CustomEvent('userflow:pushstate')
+  w.__userflowStatePatched = true
+  document.dispatchEvent(userflowPushStateEvent)
```

Comment:
Is `window` a more appropriate event target?

Reply:
I would also dispatch the event AFTER the original has been called. Just like the override we do in real Userflow.js